microprofile build doesn't work

compiler returns an error when trying to build skaled with `-DMICROPROFILE=ON`

```
/home/oleh/work/skaled/libdevcore/microprofile.cpp:986:36: error: ‘void* memset(void*, int, size_t)’ clearing an object of type ‘struct MicroProfile’ with no trivial copy-assignment; use value-initialization instead [-Werror=class-memaccess]
  986 |         memset( &S, 0, sizeof( S ) );
      |                                    ^
/home/oleh/work/skaled/libdevcore/microprofile.cpp:496:8: note: ‘struct MicroProfile’ declared here
  496 | struct MicroProfile {
      |        ^~~~~~~~~~~~
/home/oleh/work/skaled/libdevcore/microprofile.cpp: In function ‘MicroProfileThreadLog* MicroProfileCreateThreadLog(const char*)’:
/home/oleh/work/skaled/libdevcore/microprofile.cpp:1136:42: error: ‘void* memset(void*, int, size_t)’ clearing an object of type ‘struct MicroProfileThreadLog’ with no trivial copy-assignment; use value-initialization instead [-Werror=class-memaccess]
 1136 |         memset( pLog, 0, sizeof( *pLog ) );
```

--- a/libdevcore/CMakeLists.txt
+++ b/libdevcore/CMakeLists.txt
@@ -15,8 +15,9 @@ add_library(devcore ${sources} ${headers})
 add_dependencies(devcore secp256k1)
 
 target_compile_options( devcore PRIVATE
-    -Wno-error=deprecated-copy -Wno-error=unused-result -Wno-error=unused-parameter -Wno-error=unused-variable -Wno-error=maybe-uninitialized
-    )
+    -Wno-error=deprecated-copy -Wno-error=unused-result -Wno-error=unused-parameter
+    -Wno-error=unused-variable -Wno-error=maybe-uninitialized -Wno-error=class-memaccess
+)
 
 # Needed to prevent including system-level boost headers:
 target_include_directories(devcore SYSTEM PUBLIC ${Boost_INCLUDE_DIR} PRIVATE ../utils)

--- a/skaled/main.cpp
+++ b/skaled/main.cpp
@@ -550,6 +550,7 @@ int main( int argc, char** argv ) try {
     cc::_on_ = false;
     cc::_max_value_size_ = 2048;
     MicroProfileSetEnableAllGroups( true );
+    dev::setThreadName( "main" );
     BlockHeader::useTimestampHack = false;
     srand( time( nullptr ) );
     setCLocale();
@@ -2773,7 +2774,6 @@ int main( int argc, char** argv ) try {
             << cc::debug( "Done, programmatic shutdown via Web3 is disabled" );
     }
 
-    dev::setThreadName( "main" );
     if ( g_client ) {
         unsigned int n = g_client->blockChain().details().number;
         unsigned int mining = 0;

